### DISCUSSION TITLE
Synthetic Bold Text is not working properly on Linux?

### DISCUSSION THREAD

**Discussion by fuadnafiz98** [ghostty-org/ghostty#3791] [Q&A]
I am facing issues with this option `font-synthetic-style`. 

I can see from this PR (https://github.com/ghostty-org/ghostty/pull/2144) that this issue has been resolved. 

But on my linux machine, I am facing the issue with "Berkeley Mono" font. 

Am I missing something(maybe any packages) which is preventing the fonts to convert to bold / italic?

![screenshot-1735417104](https://github.com/user-attachments/assets/dac0fe9f-15bd-4cd6-8735-7bffba0539df)

**Comment by Luoyayu** [comment]
I have the same problem with the CJK fonts I use, from left to right side Gnome-Console, Ptyxis and Ghostty.
![image](https://github.com/user-attachments/assets/2dcf2fc9-370e-40ac-9302-7bd6783caf19)

The font I use is LXGW WenKai Mono
![image](https://github.com/user-attachments/assets/e4768e90-6b6f-430f-b945-28a3ee7579d8)

**Comment by renxzen** [comment]
Facing the same issue with "ProggyVector Nerd Font Mono"
Alacritty shows the bold font fine

**Comment by mbwilding** [comment]
Same issue on Arch Linux, works fine on macOS. Using my own custom font (NeoSpleen).
Will dig into the code after work and see if I can create a PR.

From what I can see, on Linux when doing font discovery, it will set Regular, Bold, Italic and Bold-Italic all to your font-family specified, so when it gets to checking if it needs to run the synth (creating fake font styles) it goes nah I'm good, we already have all the font styles needed. My assumption is on macOS, the font discovery won't do that, and will just set your Regular (if that is all you have specified, and doesn't find the matching other styles), and the synth logic will run.


I added this `continue if not regular` as a diagnostic test.
![image](https://github.com/user-attachments/assets/85b86381-69f6-475a-bd45-07966310ca08)

Can see the vim mode and clock are now bold.
![image](https://github.com/user-attachments/assets/4dd95569-a164-4bdf-91bb-1d85be4b86f6)


Original
```
info(font_shared_grid_set): font regular: NeoSpleen Nerd Font
info(font_shared_grid_set): font bold: NeoSpleen Nerd Font
info(font_shared_grid_set): font italic: NeoSpleen Nerd Font
info(font_shared_grid_set): font bold_italic: NeoSpleen Nerd Font
```

With diagnostic logic
```
info(font_shared_grid_set): font regular: NeoSpleen Nerd Font
info(font_collection): font synthetic italic created family=NeoSpleen Nerd Font
info(font_collection): synthetic italic face created
info(font_collection): font synthetic bold created family=NeoSpleen Nerd Font
info(font_collection): synthetic bold face created
info(font_collection): font synthetic bold created family=NeoSpleen Nerd Font
info(font_collection): synthetic bold italic face created from italic
```

**Comment by 00-kat** [comment]
By the way, the issue is synthetic text in general on Linux, and not *only* synthetic bold.  Some other people had issues with italics: #3588 and [some Discord] [help posts].

[some Discord]: https://discord.com/channels/1005603569187160125/1337271422782935040
[help posts]: https://discord.com/channels/1005603569187160125/1311341522951737414

**Comment by polirritmico** [comment]
Hi, this should be a bug not a discussion.

I've dig a little into this, and checked that by changing [this line](https://github.com/ghostty-org/ghostty/blob/2c06ce761b5daea6670ac78a2e2342d755705223/src/font/face/freetype.zig#L49) to true to force the italic generation, the synthetic italic works:

![imagen](https://github.com/user-attachments/assets/c239be18-7f04-4484-82ca-c480f9db18b5)

Build and system details:

```
# emerge -pv ghostty
[ebuild   R   ~] x11-terms/ghostty-1.1.0::gentoo  USE="X adwaita -man" 0 KiB
```

```
# equery depgraph ghostty
 * Searching for ghostty ...

 * dependency graph for x11-terms/ghostty-1.1.0
 `--  x11-terms/ghostty-1.1.0  ~amd64
   `--  dev-libs/oniguruma-6.9.9  (>=dev-libs/oniguruma-6.9.9) amd64
   `--  dev-util/glslang-1.3.296.0-r1  (>=dev-util/glslang-1.3.296.0) amd64
   `--  gui-libs/gtk-4.14.4-r1  (gui-libs/gtk) amd64  [X?]
   `--  media-libs/fontconfig-2.15.0-r1  (>=media-libs/fontconfig-2.14.2) amd64
   `--  media-libs/freetype-2.13.3  (>=media-libs/freetype-2.13.2) amd64  [bzip2 harfbuzz png]
   `--  media-libs/harfbuzz-10.1.0  (>=media-libs/harfbuzz-8.4.0) amd64  [truetype]
   `--  x11-libs/libX11-1.8.10-r1  (x11-libs/libX11) amd64
   `--  gui-libs/libadwaita-1.5.3  (gui-libs/libadwaita) amd64
   `--  sys-libs/ncurses-6.5_p20250125  (>=sys-libs/ncurses-6.5_p20250118) ~amd64  [-minimal]
   `--  x11-terms/ghostty-terminfo-1.1.0  (~x11-terms/ghostty-terminfo-1.1.0) [~amd64 keyword]
   `--  virtual/pandoc-0-r1  (virtual/pandoc) amd64
   `--  dev-lang/zig-0.13.0-r2  (dev-lang/zig) ~amd64  [llvm(+)]
   `--  dev-lang/zig-bin-0.13.0  (dev-lang/zig-bin) [~amd64 keyword]
   `--  virtual/pkgconfig-3  (virtual/pkgconfig) amd64
   `--  dev-util/desktop-file-utils-0.28  (dev-util/desktop-file-utils) amd64
   `--  x11-misc/shared-mime-info-2.4-r1  (x11-misc/shared-mime-info) amd64
[ x11-terms/ghostty-1.1.0 stats: packages (17), max depth (1) ]
```

For more details and a font to test it, check my first report on this: #3588

---

As a workaround I've create the italic and italic-bold variants of my FiraCode customization. Maybe they will be useful to someone: https://github.com/polirritmico/firacode-custom

**Reply by 00-kat** [reply]
Wild guess, but I think the config parser for this is broken, since in theory the `packed struct` from `Config.zig` should be passed right on to Freetype.  Perhaps someone forgot to do that or something though.

Edit: the config parsing is just a generic on `packed struct`, and I don't see any issues with the code that deals with synthetic styles...

Edit 2: the function which deals with synthetic style loading has a bunch of log messages, but I don't see a single one.
